addons/mail/static/tests/owl/component/activity_tests.js

```diff
@@ -0,0 +1,832 @@
+        1,
+        "activity should have a details button"
+    );
+
+    await activity.querySelector('.o_Activity_detailsButton').click();
+    await afterNextRender();
+    // await pause();
+    assert.strictEqual(
+        activity.querySelectorAll('.o_Activity_details').length,
+        1,
+        "activity should have details"
+    );
+
+    await activity.querySelector('.o_Activity_detailsButton').click();
+    await afterNextRender();
+    assert.strictEqual(
+        activity.querySelectorAll('.o_Activity_details').length,
+        0,
+        "activity should have no details"
+    );
+});
+
+QUnit.test('activity details layout', async function (assert) {
+    assert.expect(10);
+
+    await this.start();
+    const today = new Date();
+    const tomorrow = new Date();
+    tomorrow.setDate(today.getDate() + 1);
+    const activityLocalId = this.env.store.dispatch('_createActivity', {
+        activity_type_id: [1, 'Fake type'],
+        create_date: date_to_str(today),
+        create_uid: [1, 'Admin'],
+        date_deadline: date_to_str(tomorrow),
+        state: 'planned',
+        user_id: [1, 'Pauvre pomme'],
+    });
+    await this.createActivityComponent(activityLocalId);
+    assert.strictEqual(
+        document.querySelectorAll('.o_Activity').length,
+        1,
+        "should have activity component in DOM"
+    );
+    const activity = document.querySelector('.o_Activity');
+    assert.strictEqual(
+        activity.querySelectorAll('.o_Activity_detailsButton').length,
+        1,
+        "activity should have a details button"
+    );
+
+    await activity.querySelector('.o_Activity_detailsButton').click();
+    await afterNextRender();
+    assert.strictEqual(
+        activity.querySelectorAll('.o_Activity_details').length,
+        1,
+        "activity should have details"
+    );
+    assert.strictEqual(
+        activity.querySelectorAll('.o_Activity_details .o_Activity_type').length,
+        1,
+        "activity details should have type"
+    );
+    assert.strictEqual(
+        activity.querySelector('.o_Activity_details .o_Activity_type').textContent,
+        "Fake type",
+        "activity details type should be 'Fake type'"
+    );
+    assert.strictEqual(
+        activity.querySelectorAll('.o_Activity_detailsCreation').length,
+        1,
+        "activity details should have creation date "
+    );
+    assert.strictEqual(
+        activity.querySelectorAll('.o_Activity_detailsCreator').length,
+        1,
+        "activity details should have creator"
+    );
+    assert.strictEqual(
+        activity.querySelectorAll('.o_Activity_detailsAssignation').length,
+        1,
+        "activity details should have assignation information"
+    );
+    assert.ok(
+        activity.querySelector('.o_Activity_detailsAssignation').textContent.indexOf('Pauvre pomme') >= 0,
+        "activity details assignation information should contain creator display name"
```

Comment:
would be best to have strictEqual, here if it fails we don't know what is expected from a glance